File: webdav/rhino/update/webdav_update.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// --------------------------------------------------------------
//
//	webdav_update.js
//
//					Apr/10/2013
//
//
// --------------------------------------------------------------
importPackage (java.util);
importPackage (java.io);
importPackage (org.apache.commons.httpclient);
importPackage (org.apache.commons.httpclient.methods);

load ("/var/www/data_base/common/rhino_common/text_manipulate.js");
load ("/var/www/data_base/common/rhino_common/net_manipulate.js");
// --------------------------------------------------------------
print ("*** 開始 ***");

var id_in=arguments[0];
var population_in=arguments[1];

print	("id_in = " + id_in + "\tpopulation_in = " + population_in);

var uri_aa = "http://cddn007:3004/city/tokyo.json";

print ("uri_aa = " + uri_aa);
var 	json_str = rest_get_proc (uri_aa,"text/json");

var dict_aa =  JSON.parse (json_str);

dict_aa = dict_update_proc (dict_aa,id_in,population_in);

dict_display_proc (dict_aa);

var str_out_aa = JSON.stringify (dict_aa);

rest_put_proc (uri_aa,str_out_aa,"text/json");

print ("***** 終了 *****");
 
// --------------------------------------------------------------

File: webdav/jython/update/webdav_update.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	update/webdav_update.py
#
#					Sep/12/2013
#
# ----------------------------------------------------------------
import	sys
import	httplib
import	string
import	json
#
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_update_proc

sys.path.append ('/var/www/data_base/common/jython_common')
from jython_get_uri import get_uri_proc
from jython_get_uri import rest_put_proc
#
# ----------------------------------------------------------------
print ("*** 開始 ***")
#
key = sys.argv[1]
population_in = string.atoi (sys.argv[2])
print ("%s\t%d" % (key, population_in))
#
host = "cddn007:3004"
path = "/city/tokyo.json"
url_json = 'http://' + host + path
#
json_str = get_uri_proc (host,path)
#
#
dict_aa = json.loads (json_str)
#
dict_bb=dict_update_proc (dict_aa,key,population_in)
#
str_json = json.dumps (dict_bb)
#
rest_put_proc (url_json,str_json,"text/json")
#
print ("*** 終了 ***")
# ----------------------------------------------------------------
